tests/unit/amo/components/TestGuidesAddonCard.js

```diff
@@ -0,0 +1,239 @@
+
+    return shallowUntilTarget(
+      <GuidesAddonCard {...allProps} />,
+      GuidesAddonCardBase,
+    );
+  };
+
+  it('renders a GuidesAddonCard', () => {
+    const iconURL = 'https://addons.cdn.mozilla.net/foo.jpg';
+    const addonName = 'special-addon';
+    const addonCustomText = 'Everyone needs this cool addon.';
+    const addon = createInternalAddon({
+      ...fakeAddon,
+      icon_url: iconURL,
+      name: addonName,
+    });
+
+    const root = render({
+      addon,
+      addonCustomText,
+    });
+
+    const image = root.find('.GuidesAddonCard-content-icon');
+    expect(image).toHaveLength(1);
+    expect(image.prop('src')).toEqual(iconURL);
+    expect(image.prop('alt')).toEqual(addonName);
+
+    expect(root.find('.GuidesAddonCard-content-description')).toHaveText(
+      addonCustomText,
+    );
+    expect(root.find(AddonTitle)).toHaveLength(1);
+
+    root.setProps({ currentVersion: fakeVersion });
+    expect(root.find(AMInstallButton)).toHaveLength(1);
+  });
+
+  // TODO: This will be updated when we address the following issue:
+  // https://github.com/mozilla/addons-frontend/issues/6900.
+  it('returns null when there is no addon', () => {
+    const root = render({
+      addon: null,
+    });
+    expect(root.html()).toEqual(null);
+  });
+
+  // // TODO: We need to cover all config settings (here and on the following
+  // // test case). This will be addressed in the following issue:
+  // // https://github.com/mozilla/addons-frontend/issues/6903
+  it('renders AddonCompatibilityError when there is incompatibility', () => {
+    const root = render({
+      _getClientCompatibility: sinon.stub().returns({
+        compatible: false,
+        reason: INCOMPATIBLE_UNDER_MIN_VERSION,
+      }),
+    });
+
+    expect(root.find(AddonCompatibilityError)).toHaveLength(1);
+  });
+
+  it('does not render an AddonCompatibilityError when there is compatibility', () => {
+    const root = render({
+      _getClientCompatibility: sinon.stub().returns({
+        compatible: true,
+        reason: null,
+      }),
+    });
+
+    expect(root.find(AddonCompatibilityError)).toHaveLength(0);
+  });
+
+  it('renders Staff Pick content by default', () => {
+    const root = render();
+    expect(
+      root.find('.GuidesAddonCard-content-header-staff-pick'),
+    ).toHaveLength(1);
+  });
+
+  it('does not render Staff Pick content when staffPick prop is false', () => {
+    const root = render({ staffPick: false });
+    expect(
+      root.find('.GuidesAddonCard-content-header-staff-pick'),
+    ).toHaveLength(0);
+  });
+
+  it('passes the addon to AddonTitle', () => {
+    const addon = createInternalAddon(fakeAddon);
+    const root = render({ addon });
+
+    expect(root.find(AddonTitle)).toHaveProp('addon', addon);
+  });
+
+  it('passes the addon to AMInstallButton', () => {
+    const addon = createInternalAddon(fakeAddon);
+    const root = render({ addon });
+
+    root.setProps({ currentVersion: fakeVersion });
```

Comment:
Again, you should never pass a mapped prop (`currerntVersion`) to a component under test. You need to dispatch an action to load the version. It looks like `loadAddonResults` or `loadVersions` [will do it](https://github.com/mozilla/addons-frontend/blob/2f7083605eca96efe72d46cb57a1ad229b5cc350/src/core/reducers/versions.js#L267).

Reply:
okay I *think* I got it now. It's working now:) thanks!